src/MoviesContainer/MoviesContainer.js

```diff
@@ -10,14 +14,21 @@ function Movies({posters, changeVoteCountData, changeView}) {
         poster_path={poster.poster_path}
         vote_count={poster.vote_count}
         changeVoteCountData={ changeVoteCountData }
-        onClick={ () => changeView(poster) } 
+        onPosterSelect={ () => onPosterSelect(poster) } 
       />
     )
   })
 
+  const showDetails = (
+    <MovieDetails
+      movie={selectedMovie}
+      onBackButton={onBackButton}
+    />
+  )
+
   return (
       <section className='movies-container'>
-        {showPosters}
+        { selectedMovie ? showDetails : showPosters }
```

Comment:
The use of ternsry throughout this iteration was really appropriate, I think your implementation is well done! If this app was expanded in the future theres a potential for this to be a limiting factor, but for the project this seems like it will work perfectly!